### DISCUSSION TITLE
Reporting tools

### DISCUSSION THREAD

**Discussion by keluc** [actuarialopensource/website#37] [Team Posts]
Which setup do you use for automated analysis and reporting?

My setup is based on: 
- jupyter en nbconvert 
- nbparamaterise for passing parameters to template notebooks
- sphinx 
- nbsphinx for using notebooks in sphinx (tweaked for hiding/showing source code)
- jupytext: for version control and building notebooks from template notebooks

I've also used R-markdown and Reticulate to generation course material in R studio.

**Comment by jbogaardt** [comment]
Mine's similar though I use:
* [papermill](https://papermill.readthedocs.io/en/latest/) for notebook parameterization
* [jupyterbook](https://jupyterbook.org/intro.html) in lieu of classic sphinx
* I wrap notebook execution in [dask custom graphs](https://docs.dask.org/en/latest/custom-graphs.html) to string together dag-style workflows with parallelism.

**Comment by lewisfogden** [comment]
Not to the stage of automating this far, so I generally publish to github (private local instance), either using the ipynb renderer, or converting to html and then putting on gh pages.

Jupyterbook looks very impressive - I'm going to try that out now!
